doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,951 @@
+* Extend MODULE_TYPE in ModuleBase class with MODULE_TYPE_DPU and MODULE_TYPE_SWITCH to support SmartSwitch
+
+#### 3.1.1 ChassisBase class API enhancements
+is_modular_chassis(self):
+```
+    Retrieves whether the sonic instance is part of modular chassis. For SmartSwitch platforms this should return True even if they are
+    fixed-platforms, as they are treated like a modular chassis as the DPU cards are treated like line-cards of a modular-chassis.
+
+    Returns:
+      False
+```
+
+get_num_modules(self):
+```
+    Retrieves the number of modules available on this chassis including DPUs.
+    This is an existing API.
+
+    Returns:
+        An integer, the number of modules available on this chassis
+```
+
+get_all_modules(self):
+```
+    Retrieves all modules available on this chassis including DPUs
+
+    Returns:
+        A list of objects derived from ModuleBase representing all modules
+        available on this chassis
+```
+
+get_module(self, index):
+```
+    Retrieves module represented by (0-based) index <index>
+    On a SmartSwitch index:0 will fetch switch, index:1 will fetch
+    DPU0 and so on
+
+    Args:
+        index: An integer, the index of the module to retrieve
+
+    Returns:
+        An object derived from ModuleBase representing the specified module
+```
+
+get_module_index(self, module_name):
+```
+    Retrieves module index from the module name
+
+    Args:
+        module_name: A string, Ex. SWITCH, DPU1, DPU2 ... DPUX
+
+    Returns:
+        An integer, the index of the ModuleBase object in the module_list
+```
+#### 3.1.2 ChassisBase class new APIs
+The DPU ID is used only for indexing purpose and is internal to platform.  There is a dpu_id to dpu_name conversion API and the dpu_name is the one that is used externally in all CLIs.
+
+get_dpu_id(self, name):
+```
+    Retrieves the DPU ID for the given dpu-module name. Returns None for non-smartswitch chassis.
+
+    Returns:
+        An integer, indicating the DPU ID Ex: name:DPU0 return value 1, name:DPU1 return value 2, name:DPUX return value X+1
+```
+
+is_smartswitch(self):
+```
+    Retrieves whether the sonic instance is part of smartswitch
+
+    Returns:
+      True
+```
+
+get_module_dpu_data_port(self, index):
+```
+    Retrieves the DPU data port NPU-DPU association represented for the DPU index. Platforms that need to overwrite the platform.json file will use this API
+
+    This is valid only on the Switch and not on DPUs
+
+    Args:
+        index: An integer, the index of the module to retrieve
+
+    Returns:
+        NPU-DPU port association: A string Ex: For index: 1 will return the dup0 port association which is "Ethernet-BP0: Ethernet0" where the string left of ":" (Ethernet-BP0) is the NPU port and the string right of ":" (Ethernet0) is the DPU port.
+```
+#### 3.1.3 NPU to DPU data port mapping
+platform.json of NPU/switch will show the NPU to DPU data port mapping. This will be used by services early in the system boot. 
+```
+{
```

Comment:
We agreed to have the following schema for this:
```
"DPUs" : [
    {
      "dpu0": {
                "interface": {"Ethernet224": "Ethernet0"}
       }
    },
    {
       "dpu1": {
                "interface": {"Ethernet232": "Ethernet0"}
        },
    },
    .
    .
    {
       "dpuX": {
                "interface": {"EthernetX": "EthernetY"}
        }
    }
]
```
Having the mapping under the `interface` key will simplify the implementation as there will be multiple attributes for each DPU.
![image](https://github.com/sonic-net/SONiC/assets/17766225/a32f57d2-2429-487c-ad7c-7abb978a60b9)

Reply:
Fixed it is as shown here .